### ISSUE TITLE
Loss becoming nan

### ISSUE THREAD

**Issue by 1774537072** [Zeng-Shuang/FedLD#1]
Hi, thank you for sharing your great work!

I'm interested in your SVD method for global aggregation weights. However, when I tried to reproduce it in other tasks, specifically in Federated Learning for Brain Tumor Segmentation, I encountered issues with the loss becoming NaN (weighted cross-entropy loss and Dice loss). Sometimes it happens in round 200+, and sometimes in round 2 or 8.

Do you have any clue about this problem?

**Comment by Zeng-Shuang** [comment]
Sorry for the late response. We have encountered this problem before, it shows that it is important to choose a proper learning rate. Maybe you can try different learning rates for local training. You can also try different number of local epochs in each round. Hope this will be helpful for solving the nan problem in your tasks.

**Comment by 1774537072** [comment]
Thank you for your response, I finally resolved the problem. However, I have another question:
In the third step (3) of the figure 1 annotation, it says "After that, each client uploads its local gradient to the global server." But in the code file “running.py”, function "train_round_fedld" is written like this:
```
for idx in idx_users:
    local_clients_grads.append(get_grads_(local_clients[idx].local_model, global_model))
```
Isn't this uploading model parameters and using the parameters to obtain gradients on the server?
Please correct me if I am wrong.
